src/smtp.rs

```diff
@@ -529,9 +534,12 @@ pub(crate) async fn send_smtp_messages(context: &Context, connection: &mut Smtp)
             .context("failed to send message")?;
     }
 
-    let ratelimited = send_mdns(context, connection)
-        .await
-        .context("failed to send MDNs")?;
+    if !ratelimited {
```

Comment:
but we still need the return value to be `true` as otherwise the following may happen:
- webxdc-updates are not sent out due to ratelimit
- sending the other messages takes some time
- mdns are send out as ratelimit expired in between
in this case, the caller will not wait for `until_can_send` and the webxdc updates are not send in time. also, this would send mdn with a sometimes higher priority than webxdc updates.

Reply:
i added a comment to the source code.